src/main/java/dk/cs/aau/huppaal/presentations/LogTabPresentation.java

```diff
@@ -0,0 +1,175 @@
+                // Try to open the link
+                default -> {
+                    var f = new File(ref);
+                    if(f.exists())
+                        Desktop.getDesktop().open(f);
+                    else
+                        Desktop.getDesktop().browse(new URI(ref));
+                }
+            }
+        } catch (Exception e) {
+            HUPPAAL.showToast(e.getMessage());
+            Log.addError(e.getMessage());
+        }
+    }
+
+    private void onLogAdded(Log log) {
+        if(!log.level().equals(controller.level))
+            return;
+        var logMessage = log.message();
+        var matcher = LogRegex.PATTERN.matcher(log.message());
+        var index = 0;
+        while(matcher.find()) {
+            logArea.appendText(logMessage.substring(index, matcher.start()));
+            index = matcher.end();
+            var q = LogLinkQuantifier.valueOf(matcher.group("quantifier").toUpperCase());
+            logArea.appendWithLink(matcher.group("display"), matcher.group(), q);
+        }
+        logArea.appendText(logMessage.substring(index) + "\n");
+    }
+
+    private void initializeLogArea(String textColor) {
+        logArea.setTextInsertionStyle(new TextStyle().updateTextColorWeb(textColor));
+        logArea.getStyleClass().add("log-text");
+        logArea.setEditable(false);
+        controller.logAreaInsertionPoint.getChildren().add(scrollPane);
+    }
+
+    private void initializeButtons() {
+        Tooltip.install(controller.clearLogsButton, new Tooltip("Clear log"));
+        controller.clearLogsButton.setOnMouseClicked(e -> {
+            Log.clearLogsForLevel(controller.level); // This is just to be polite to the logging framework, and your RAM
+            logArea.clear();
+        });
+
+        Tooltip.install(controller.wrapTextButton, new Tooltip("Toggle word wrap"));
+        controller.wrapTextButton.setOnMouseClicked(e -> {
+            wordwrap = !wordwrap;
+            setupWordWrap();
+        });
+
+        Tooltip.install(controller.autoscrollLogButton, new Tooltip("Toggle autoscroll"));
+        controller.autoscrollLogButton.setOnMouseClicked(e -> toggleAutoScroll());
+    }
+
+    private void toggleAutoScroll() {
+        setAutoscroll(!autoscroll);
+    }
+
+    private void setAutoscroll(boolean value) {
+        autoscroll = value;
+        setupAutoscroll();
+    }
+
+    private final ChangeListener<String> logAreaChangeEventListener = (e,o,n) -> {
+        if(scrollPane == null)
+            return;
+        Platform.runLater(this::scrollToLastLine);
+    };
+
+    private void setupAutoscroll() {
+        if(autoscroll) {
+            controller.autoscrollLogButtonIcon.setIconLiteral("gmi-playlist-add-check");
+            controller.autoscrollLogButton.setStyle("-fx-background-color: rgba(255,255,255,0.1)");
+            logArea.textProperty().addListener(logAreaChangeEventListener);
```

Comment:
Will this logic not add a listener to the text property every time the auto-scroll is directly set? We have run into a problem with ECDAR where we end up with 400+ listeners after a few repeated clicks in the right order

Reply:
Update: I fixed it again in 35096883dedc1e30a9a42af16dcdd7669c12dcdb